flake.nix

```diff
@@ -32,15 +32,47 @@
           z3
           cvc5
           git
+        ] ++ lib.optionals stdenv.isDarwin [
           foundry.defaultPackage.${system}
         ];
 
         secp256k1-static = stripDylib (pkgs.secp256k1.overrideAttrs (attrs: {
           configureFlags = attrs.configureFlags ++ [ "--enable-static" ];
         }));
 
+        patchedHaskellPackages = pkgs.haskell.packages.ghc94.override {
+          overrides = self: super: {
+            # disable tests in optics
+            optics-core = pkgs.haskell.lib.dontCheck (self.callCabal2nix "optics-core" (pkgs.fetchFromGitHub {
+              owner = "well-typed";
+              repo = "optics";
+              rev = "46b03019bd7d9eddb767b19a68b94125eec3b17a";
+              sha256 = "sha256-wzZ/64G7nVAzIFjKNs/jBvv6gQdTIQS4X/OvM5KWfnU=";
+            } + "/optics-core") {});
+            optics = pkgs.haskell.lib.dontCheck (self.callCabal2nix "optics" (pkgs.fetchFromGitHub {
+              owner = "well-typed";
+              repo = "optics";
+              rev = "46b03019bd7d9eddb767b19a68b94125eec3b17a";
+              sha256 = "sha256-wzZ/64G7nVAzIFjKNs/jBvv6gQdTIQS4X/OvM5KWfnU=";
+            } + "/optics") {});
+            indexed-profunctors = pkgs.haskell.lib.dontCheck (self.callCabal2nix "optics" (pkgs.fetchFromGitHub {
+              owner = "well-typed";
+              repo = "optics";
+              rev = "46b03019bd7d9eddb767b19a68b94125eec3b17a";
+              sha256 = "sha256-wzZ/64G7nVAzIFjKNs/jBvv6gQdTIQS4X/OvM5KWfnU=";
+            } + "/indexed-profunctors") {});
+            # use obsidian systems fork of string-qq
+            string-qq = self.callCabal2nix "string-qq" (pkgs.fetchFromGitHub {
```

Comment:
can we upstream this to nixpkgs?

Reply:
Sure, I'll make a PR to Nixpkgs explaining the situation with string-qq